akka-distributed-data/src/main/scala/akka/cluster/ddata/protobuf/ReplicatorMessageSerializer.scala

```diff
@@ -80,9 +171,9 @@ class ReplicatorMessageSerializer(val system: ExtendedActorSystem)
 
   def toBinary(obj: AnyRef): Array[Byte] = obj match {
     case m: DataEnvelope   ⇒ dataEnvelopeToProto(m).toByteArray
-    case m: Write          ⇒ writeToProto(m).toByteArray
-    case WriteAck          ⇒ dm.Empty.getDefaultInstance.toByteArray
-    case m: Read           ⇒ readToProto(m).toByteArray
+    case m: Write          ⇒ writeCache.getOrAdd(m, () ⇒ writeToProto(m).toByteArray)
```

Comment:
Instead of allocating this function on each Write, why not prove the function when constructing the SmallCache?